Cant compile on Arch using Kernel 6.1

[cryptodan@alphacentauri ~]$ git clone https://github.com/McMCCRU/rtl8188gu.git
Cloning into 'rtl8188gu'...
remote: Enumerating objects: 838, done.
remote: Counting objects: 100% (30/30), done.
remote: Compressing objects: 100% (19/19), done.
remote: Total 838 (delta 14), reused 18 (delta 11), pack-reused 808
Receiving objects: 100% (838/838), 2.98 MiB | 18.73 MiB/s, done.
Resolving deltas: 100% (432/432), done.
[cryptodan@alphacentauri ~]$ cd rtl8188gu/
[cryptodan@alphacentauri rtl8188gu]$ make 
make ARCH=x86_64 CROSS_COMPILE= -C /lib/modules/6.1.1-arch1-1/build M=/home/cryptodan/rtl8188gu  modules
make[1]: Entering directory '/usr/lib/modules/6.1.1-arch1-1/build'
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_cmd.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_security.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_debug.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_io.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_ioctl_query.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_ioctl_set.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_ieee80211.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_mlme.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_mlme_ext.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_mi.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_wlan_util.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_vht.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_pwrctrl.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_rf.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_recv.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_sta_mgt.o
/home/cryptodan/rtl8188gu/core/rtw_sta_mgt.c: In function ‘rtw_mfree_stainfo’:
/home/cryptodan/rtl8188gu/core/rtw_sta_mgt.c:352:25: warning: the comparison will always evaluate as ‘true’ for the address of ‘lock’ will never be NULL [-Waddress]
  352 |         if (&psta->lock != NULL)
      |                         ^~
In file included from /home/cryptodan/rtl8188gu/include/drv_types.h:106,
                 from /home/cryptodan/rtl8188gu/core/rtw_sta_mgt.c:17:
/home/cryptodan/rtl8188gu/include/sta_info.h:229:17: note: ‘lock’ declared here
  229 |         _lock   lock;
      |                 ^~~~
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_ap.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_xmit.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_p2p.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_rson.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_tdls.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_br_ext.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_iol.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_sreset.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_btcoex_wifionly.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_btcoex.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_beamforming.o
  CC [M]  /home/cryptodan/rtl8188gu/core/rtw_odm.o
  CC [M]  /home/cryptodan/rtl8188gu/core/efuse/rtw_efuse.o
  CC [M]  /home/cryptodan/rtl8188gu/os_dep/osdep_service.o
/home/cryptodan/rtl8188gu/os_dep/osdep_service.c: In function ‘rtw_random32’:
/home/cryptodan/rtl8188gu/os_dep/osdep_service.c:2509:16: error: implicit declaration of function ‘prandom_u32’; did you mean ‘prandom_u32_max’? [-Werror=implicit-function-declaration]
 2509 |         return prandom_u32();
      |                ^~~~~~~~~~~
      |                prandom_u32_max
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:250: /home/cryptodan/rtl8188gu/os_dep/osdep_service.o] Error 1
make[1]: *** [Makefile:1992: /home/cryptodan/rtl8188gu] Error 2
make[1]: Leaving directory '/usr/lib/modules/6.1.1-arch1-1/build'
make: *** [Makefile:1895: modules] Error 2

--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -1628,7 +1628,9 @@ exit:
 }
 
 static int cfg80211_rtw_add_key(struct wiphy *wiphy, struct net_device *ndev,
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) || defined(COMPAT_KERNEL_RELEASE)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)) || defined(COMPAT_KERNEL_RELEASE)
+    int link_id, u8 key_index, bool pairwise, const u8 *mac_addr,
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37))
 	u8 key_index, bool pairwise, const u8 *mac_addr,
 #else	/* (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) */
 	u8 key_index, const u8 *mac_addr,
@@ -1765,7 +1767,9 @@ addkey_end:
 }
 
 static int cfg80211_rtw_get_key(struct wiphy *wiphy, struct net_device *ndev,
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) || defined(COMPAT_KERNEL_RELEASE)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)) || defined(COMPAT_KERNEL_RELEASE)
+    int link_id, u8 key_index, bool pairwise, const u8 *mac_addr,
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37))
 	u8 key_index, bool pairwise, const u8 *mac_addr,
 #else	/* (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) */
 	u8 key_index, const u8 *mac_addr,
@@ -1797,7 +1801,9 @@ static int cfg80211_rtw_get_key(struct wiphy *wiphy, struct net_device *ndev,
 }
 
 static int cfg80211_rtw_del_key(struct wiphy *wiphy, struct net_device *ndev,
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) || defined(COMPAT_KERNEL_RELEASE)
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)) || defined(COMPAT_KERNEL_RELEASE)
+                int link_id, u8 key_index, bool pairwise, const u8 *mac_addr)
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37))
 				u8 key_index, bool pairwise, const u8 *mac_addr)
 #else	/* (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 37)) */
 				u8 key_index, const u8 *mac_addr)
@@ -1817,9 +1823,11 @@ static int cfg80211_rtw_del_key(struct wiphy *wiphy, struct net_device *ndev,
 }
 
 static int cfg80211_rtw_set_default_key(struct wiphy *wiphy,
-	struct net_device *ndev, u8 key_index
-	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 38)) || defined(COMPAT_KERNEL_RELEASE)
-	, bool unicast, bool multicast
+	struct net_device *ndev,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0)) || defined(COMPAT_KERNEL_RELEASE)
+    int link_id, u8 key_index, bool unicast, bool multicast
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(2, 6, 38))
+    u8 key_index, bool unicast, bool multicast
 	#endif
 )
 {

--- a/os_dep/linux/os_intfs.c
+++ b/os_dep/linux/os_intfs.c
@@ -1594,7 +1594,11 @@ int rtw_os_ndev_register(_adapter *adapter, const char *name)
 	u8 rtnl_lock_needed = rtw_rtnl_lock_needed(dvobj);
 
 #ifdef CONFIG_RTW_NAPI
-	netif_napi_add(ndev, &adapter->napi, rtw_recv_napi_poll, RTL_NAPI_WEIGHT);
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	netif_napi_add_weight(ndev, &adapter->napi, rtw_recv_napi_poll, RTL_NAPI_WEIGHT);
+#else
+    netif_napi_add(ndev, &adapter->napi, rtw_recv_napi_poll, RTL_NAPI_WEIGHT);
+#endif
 #endif /* CONFIG_RTW_NAPI */
 
 #if defined(CONFIG_IOCTL_CFG80211)

--- a/os_dep/osdep_service.c
+++ b/os_dep/osdep_service.c
@@ -2505,7 +2505,9 @@ u64 rtw_division64(u64 x, u64 y)
 inline u32 rtw_random32(void)
 {
 #ifdef PLATFORM_LINUX
-#if (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+    return get_random_u32(); //prandom_u32 was just a wrapper around get_random_u32
+#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(3, 8, 0))
 	return prandom_u32();
 #elif (LINUX_VERSION_CODE <= KERNEL_VERSION(2, 6, 18))
 	u32 random_int;

